site/routes/definitions.js

```diff
@@ -1,14 +1,27 @@
 // @flow
 
-type RouteDefinition = {|
+type RouteDefinition = {
   title: string | (props: Object) => string,
   key: string,
   route: string,
-  defaults?: any,
   stateToProps?: (state: Object, params?: Object) => any,
   gen?: (state: Object) => Array<Object>,
-  render?: (state: Object) => any,
-|}
+}
+
+const getBadgersByCategory = (badgers, category) => (
+  badgers.filter(badger => badger.categories.filter(cat => cat === category).length > 0)
+);
+
+const createPages = (category, count) => (
+  Array.from(Array(Math.ceil(count / 20)).keys()).map(page => ({ category, page: page + 1 }))
+);
+
+const genBadgersParams = state => (
+  state.categories.reduce((params, category) => {
+    const count = getBadgersByCategory(state.badgers, category).length;
+    return params.concat(createPages(category.toLowerCase(), count));
+  }, createPages('everyone', state.badgers.length + 1))
+);
```

Comment:
I think this logic belongs somewhere else - could it go in the fetcher? or a different file in the `state/` folder providing convenience functions for accessing that particular part of state?